libraries/Microsoft.Bot.Builder/DictionaryExtensions.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder
+{
+    /// <summary>
+    /// Extension method on IDictionary <see cref="IDictionary"/>.
+    /// </summary>
+    public static partial class DictionaryExtensions
+    {
+        private static JsonSerializer _coerceSerializer = new JsonSerializer() { TypeNameHandling = TypeNameHandling.None };
+
+        /// <summary>
+        /// Extension Method on IDictionary to coerce the value of the dictionary to type T.
+        /// </summary>
+        /// <remarks>This allows the dictionaries which are string => JToken to be able to affinitize to a typed instance.
+        /// If the object in the dictionary is not currently of type <typeparamref name="T"/>, it will be replaced with
+        /// a new object which has been casted to <typeparamref name="T"/>.</remarks>
+        /// <param name="dict"><see cref="IDictionary"/> from which to retrieve the property.</param>
+        /// <param name="property">Name of the property.</param>
+        /// <typeparam name="T">The type to which the object should be casted.</typeparam>
+        /// <returns>The in from the dictionary casted to <typeparamref name="T"/>.</returns>
+        public static T MapValueTo<T>(this IDictionary<string, object> dict, string property)
```

Comment:
Can we rename this method to GetValue<T>?  MapTo is a new term and donet uses GetValue<T> on [JsonNode](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.nodes.jsonnode.getvalue?view=net-6.0) and we also use GetValueAsync<T> in our own `IStatePropertyAccessor`
```suggestion
        public static T GetValue<T>(this IDictionary<string, object> dict, string property)
```

Reply:
This method actually alters the reference and type in the underlying dictionary. It feels like the name should reflect the fact there are side effects.